replay-assets/replay_command_handlers.js

```diff
@@ -460,6 +460,9 @@ function Target_getCurrentNetworkStreamData(params) {
 
 function Target_topFrameLocation() {
   try {
+    if (!isReplayScriptAlive()) {
+      return {};
```

Comment:
I wonder if it would be helpful to return a better error message to the client. For instance, I don't think we want an empty location or stack, we want to know that we were asking for one at the wrong time.

Reply:
Also - does it make sense to write a new runtime regression test for this case

Reply:
1. We don't have a way of communicating a "why" to the client with the current protocol.
2. We don't know how to test this type of lifecycle-dependent situation, since we have no way, from the outside, to determine what "liveness" state we are in, nor to guarantee that the message will be processed before the liveness state changes again.
   * For example, we can't put it in the page itself (like we did with the [recorder-only command test](https://github.com/replayio/backend/pull/9606#pullrequestreview-1899758308)), since the page is shutdown at the time when we want to test this.
   * We would need some sort of additional messaging to observe the event, so as to send a request on time, but there is no guarantee that liveness has not changed again by the time our request is processed.

Reply:
or a method to force the JS into that state.  That's probably the way to go.  Then we can have a series of tests against the command handling in both active/shutting-down states.

Reply:
I think what Chris is proposing is a unit test that mocks out `isReplayScriptAlive()` and allows you to unit test `Target_topFrameLocation`?